src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (manufacturer_id, model) VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement creatStatement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            creatStatement.setLong(1, car.getManufacturer().getId());
+            creatStatement.setString(2, car.getModel());
+            creatStatement.executeUpdate();
+            ResultSet generatedKeys = creatStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create car: " + car, e);
+        }
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT cars.id, cars.manufacturer_id, cars.model, manufacturers.name, "
+                + "manufacturers.country "
+                + "FROM cars "
+                + "INNER JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.id = ? AND cars.is_deleted = FALSE;";
+        Car car = new Car();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatment = connection.prepareStatement(query)) {
+            getStatment.setLong(1, id);
+            ResultSet resultSet = getStatment.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id, e);
+        }
+        List<Driver> driversFromCar = getDriversFromCar(id);
+        car.setDrivers(driversFromCar);
+        return Optional.ofNullable(car);
```

Comment:
Instead of returning an empty Optional if the car is not found, it would be better to return Optional.empty(). This makes it clearer that no value is present.